### PR TITLE
add push docker image workflow

### PR DESCRIPTION
**Pull request by area363**
I tested the code on my origin repository and then changed the organization name in the yml file to "planetariumhq".

### CODE REVIEW

File: .github/workflows/push_docker_image.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      - name: checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: login
+        run: docker login --username ${{ secrets.DOCKER_USERNAME }} --password ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+      - name: build
+        run: docker build . -t planetariumhq/ninechronicles-headless:git-${{ github.sha }} --build-arg COMMIT=git-${{ github.sha }}
+      - name: push (publish)
+        run: docker push planetariumhq/ninechronicles-headless:git-${{ github.sha }}
```
Comment (limebell) [inline .github/workflows/push_docker_image.yml:21]:
```suggestion
        run: docker push planetariumhq/ninechronicles-headless:git-${{ github.sha }}

```

File: .github/workflows/push_docker_image.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+jobs:
+  build_and_push:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: login
+        run: docker login --username ${{ secrets.DOCKER_USERNAME }} --password ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
```
Comment (riemannulus) [inline .github/workflows/push_docker_image.yml:17]: Have we added these secrets to this upstream repository?

Comment (dahlia) [reply .github/workflows/push_docker_image.yml:17]: We should add them *before* merging this.

Comment (dahlia) [reply .github/workflows/push_docker_image.yml:17]:
```suggestion
        run: docker login --username '${{ secrets.DOCKER_USERNAME }}' --password '${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}'
```

Comment (area363) [reply .github/workflows/push_docker_image.yml]: ok i'll add the secrets right now.

**Comment by area363** [comment]
@planetarium/libplanet Secrets have been added to the upstream repository.

**Review by dahlia — APPROVED** [review]

**Comment by area363** [comment]
@planetarium/libplanet I updated the docker workflow to publish git-commit versions as well as the "latest" version so that it would be more convenient for users.

File: .github/workflows/push_docker_image.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: login
+        run: docker login --username '${{ secrets.DOCKER_USERNAME }}' --password '${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}'
+      - name: build
+        run: docker build . -t planetariumhq/ninechronicles-headless:git-${{ github.sha }} --build-arg COMMIT=git-${{ github.sha }} -t planetariumhq/ninechronicles-headless:latest --build-arg COMMIT=git-${{ github.sha }}
```
Comment (moreal) [inline .github/workflows/push_docker_image.yml:19]: I had thought since I started to review this PR. The build argument `COMMIT` is used to specify the version suffix of `NineChronicles.Standalone.Executable`. I wonder we need to specify the versioning policy of it.
When I read [https://semver.org](https://semver.org) again, it seems to present build metadata in [another way](https://semver.org/lang/ko/#spec-item-10) but ours haven't. So I thought it will be good that we should discuss it but you can feel that it is early or doesn't need to concern about it. I wonder your opinions @planetarium/libplanet.

Comment (moreal) [inline .github/workflows/push_docker_image.yml:19]: Though this comment is not related to the above comment, does it need to pass the same build argument twice? 🤔

**Review by moreal — APPROVED** [review]

File: .github/workflows/push_docker_image.yml (hunk 3 above)
Comment (area363) [reply .github/workflows/push_docker_image.yml:19]: @moreal It does appear alittle redundant 😅. We could adjust the Dockerfile so it doesn't require a version suffix as a build argument but I'll bring this issue up tomorrow!

Comment (longfin) [reply .github/workflows/push_docker_image.yml:19]: I guess it would be better if we changed version suffix from `1.0.0-<GIT-SHA>` to `1.0.0+<GIT-SHA>` to follow semver. (even if another PR)

**Review by limebell — APPROVED** [review]

**Review by longfin — APPROVED** [review]
